### DISCUSSION TITLE
Expo Map WMS Overlay

### DISCUSSION THREAD

**Discussion by anujdivesh** [expo/expo#14339] [Archived: Questions]
Hi:
I was wondering if we can overlay WMS layers on react maps. I was following example on: https://github.com/react-native-maps/react-native-maps/tree/master/example/examples 

 I was not able to do this with my own WMS URL

Thanks,

**Comment by RyanTG** [comment]
You said "Expo Map" and "React Maps", but I'm assuming you're referring to react-native-maps. It is possible: https://github.com/react-native-maps/react-native-maps#using-a-custom-tile-overlay

You'll still need to include a google maps key on android. Your tiles will load over the google maps tiles.

**Reply by anujdivesh** [reply]
Hi Much appreciate your quick response. Do we need the keys also for iOS device as well? Also if there is a better example where a layer from web mapping service was over-layed on google map.
Thanks

**Reply by RyanTG** [reply]
If you are using Apple Maps on the iOS device, you do not need a key for Apple Maps (it's free, unlimited use). If you are using Google Maps for both Android and iOS, then the key you supply will work for both. Though, if I'm remembering correctly, when you create a Google Maps API key, you need to specify where you will be using it, but I can't remember if you need to simply specify "mobile" or both "Android & iOS".

I briefly used tiles back in 2019 like this (I used OSM for testing, but eventually I used my own server). I did something like:

````
import MapView, { UrlTile } from 'react-native-maps'
...
                        <UrlTile
                            urlTemplate={`http://a.tile.openstreetmap.org/{z}/{x}/{y}.png`}
                            maximumZ={20}
                        />
````

And my app.json included things like:
````
    "ios": {
      "config": {
        "googleMapsApiKey": "key",
      },
    },
    "android": {
      "config": {
        "googleMaps": {
          "apiKey": "key"
        },
        "googleSignIn": {
          "apiKey": "key"
        }
      },
````

**Reply by anujdivesh** [reply]
Looks cool. I think ill get a google maps key and give it a shot. Just a very last question: is it possible to have a time series control on the map, where users can render the Tile layers as we slide or pause/play the time series control. Normally used for forecast weather data. Thanks in advance.

**Reply by RyanTG** [reply]
Not sure. I'd check the react-native-maps docs or issues for things about that.
